homeassistant/components/plex/update.py

```diff
@@ -0,0 +1,81 @@
+"""Representation of Plex updates."""
+import logging
+from typing import Any
+
+from plexapi.exceptions import PlexApiException
+import plexapi.server
+import requests.exceptions
+
+from homeassistant.components.update import UpdateEntity, UpdateEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import CONF_SERVER_IDENTIFIER
+from .helpers import get_plex_server
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Plex media_player from a config entry."""
+    server_id = config_entry.data[CONF_SERVER_IDENTIFIER]
+    server = get_plex_server(hass, server_id)
+    plex_server = server.plex_server
+    async_add_entities([PlexUpdate(plex_server)], update_before_add=True)
+
+
+class PlexUpdate(UpdateEntity):
+    """Representation of a Plex server update entity."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL | UpdateEntityFeature.RELEASE_NOTES
+    )
+
+    def __init__(self, plex_server: plexapi.server.PlexServer) -> None:
+        """Initialize the Update entity."""
+        self.plex_server = plex_server
+        self.can_update: bool = False
+        self._attr_name = f"Plex Media Server ({plex_server.friendlyName})"
+        self._attr_unique_id = plex_server.machineIdentifier
+        self._release_notes: str | None = None
+
+    def update(self) -> None:
+        """Update sync attributes."""
+        self._attr_installed_version = self.plex_server.version
+        try:
+            if (release := self.plex_server.checkForUpdate()) is None:
+                return
+            self.can_update = self.plex_server.canInstallUpdate()
+        except (requests.exceptions.RequestException, PlexApiException):
+            _LOGGER.warning("Polling update sensor failed, will try again")
```

Comment:
This will log this warning each time. Preferably this should only log once.

Maybe this is very well fitted to use a DataUpdateCoordinator?

Reply:
Downgraded to debug as the server being unreachable will likely be obvious elsewhere.